AirCasting/CreateSessionViews/TurnOnLocation/TurnOnLocationViewModel.swift

```diff
@@ -4,62 +4,64 @@
 import Foundation
 import Resolver
 
+enum TurnOnLocationUiProcess {
+    case sdClear
+    case sdSync
+    case createSession(sessionContext: CreateSessionContext)
+}
+
 // [RESOLVER] Init this VM inside View
 class TurnOnLocationViewModel: ObservableObject {
     @Published var isPowerABLinkActive = false
     @Published var isTurnBluetoothOnLinkActive = false
     @Published var isMobileLinkActive = false
     @Published var restartABLink = false
+    @Published var unplugABLink = false
     @Published var alert: AlertInfo?
-    var isSDClearProcess: Bool
-
+    
     @Injected private var locationAuthorization: LocationAuthorization
     @Injected private var bluetoothHandler: BluetoothPermisionsChecker
-    private let sessionContext: CreateSessionContext
     
-    var shouldShowAlert: Bool {
-        locationAuthorization.locationState == .denied
-    }
+    private let process: TurnOnLocationUiProcess
     
-    var isMobileSession: Bool {
-        sessionContext.sessionType == .mobile
+    init(process: TurnOnLocationUiProcess) {
+        self.process = process
     }
     
-    var getSessionContext: CreateSessionContext {
-        sessionContext
+    var isSDClearProcess: Bool {
+        return if case .sdClear = process { true } else { false }
     }
     
-    init(sessionContext: CreateSessionContext, isSDClearProcess: Bool) {
-        self.sessionContext = sessionContext
-        self.isSDClearProcess = isSDClearProcess
+    var shouldShowAlert: Bool {
+        locationAuthorization.locationState == .denied
     }
     
     func requestLocationAuthorisation() {
         locationAuthorization.requestAuthorization()
     }
     
-    func checkIfBluetoothDenied() -> Bool {
-       bluetoothHandler.isBluetoothDenied()
-    }
-    
     func onButtonClick() {
-        switch shouldShowAlert {
-        case true:
-            alert = InAppAlerts.locationAlert()
-        case false:
-            locationOnStep()
+        if shouldShowAlert {
+            showRequestLocationAlert()
+        } else {
+            switch process {
+            case .sdClear:
+                restartABLink.toggle()
+            case .sdSync:
+                unplugABLink.toggle()
+            case .createSession(let sessionContext):
+                if sessionContext.isMobileSession {
+                    isMobileLinkActive = true
+                } else if bluetoothHandler.isBluetoothDenied() {
+                    isTurnBluetoothOnLinkActive = true
+                } else {
+                    isPowerABLinkActive.toggle()
+                }
+            }
```

Comment:
This looks weird, but seemingly that's how the old logic worked, so I didn't want to change it, fearing something might break

Reply:
Ah, yes, the famous last words.

Reply:
I retraced this logic just to double check and everything actually looks good, though I have no idea where unplugABLink.toggle() came form :D